bzhang - bootcamp

ben-zhang opened the pull request.

ben-zhang committed:
Hypothetically Completed Bootcamp

uwuarg commented:
Build finished.

benjaminwinger commented on modules/core/include/merge_sort.h:
```diff
@@ -0,0 +1,31 @@
+ * @param lo
+ * @param hi
+ */
+void merge_sort(std::vector<int> a, int lo, int hi);
```
This doesn't seem to match the implementation

benjaminwinger commented on modules/core/include/merge_sort.h:
```diff
@@ -0,0 +1,31 @@
+/**
+ * @file contour_comparison.h
+ * @author Ben Zhang
```
In general we put WARG as the author in the files and let git track individual authorship. While this does work for a file you've just created, it's annoying to maintain, especially since we have full change and author history automatically through git.

benjaminwinger reviewed (changes requested):
Try to add changes individually rather than adding the entire tree, that way you avoid adding unwanted files (e.g. ".idea"). You can also add patterns to the global git ignore to avoid adding files specific to your workspace. (https://help.github.com/articles/ignoring-files/).

In general this looks good, though I've noted a couple issues in comments. two of which should be immediately apparent if you try to compile the project.

ben-zhang commented:
I'm having trouble compiling (in CLion), so I wrote most of the code using dev c++ instead.  When I open the project in CLion, the error "cannot find "findopencv.cmake"" is thrown, even when opening the original master version of the bootcamp. I've looked around online for a solution, but haven't found a conclusive one. Has the error occurred before in previous bootcamps?

benjaminwinger commented:
I believe that error means that CMake cannot detect the OpenCV installation directory (OpenCV installs include OpenCVConfig.cmake files that describe how it is installed). Where/how did you install OpenCV? There is also an OPENCV_DIR environment variable that I believe CMake can use to find the OpenCV installation directory.

ben-zhang commented:
I have OpenCV installed in the top level of my C drive, as in C:\opencv (I'm on Windows). I have Cygwin installed in the same place, so I figured Cmake would be able to detect it, but I may have done something wrong.

benjaminwinger commented:
I'm not entirely sure how CMake determines the installation location of opencv in windows (again, I have seen reference to an OPENCV_DIR environment variable). You can also try using docker to get a linux environment to build it with, we have a docker image you can download (instructions [here](http://docs.uwarg.com/computer-vision/Installation/))

ben-zhang commented:
Hey,

I've set the OpenCV_DIR variable to where it should go, and it seems to be working fine. But now, the following error displays:

CMake Error at CMakeLists.txt:9 (find_package):
  Found package configuration file:

    c:/opencv/build/OpenCVConfig.cmake

  but it set OpenCV_FOUND to FALSE so package "OpenCV" is considered to be
  NOT FOUND.

ben-zhang committed:
wore

uor

ben-zhang committed:
Author and Implementation

Altered Author names and implementaiton of mergeSort in the merge_sort.h header file.

ben-zhang committed:
added executable file to test unit tests

ben-zhang committed:
Added semicolon

ben-zhang committed:
fix

ben-zhang committed:
Fixed naming issue

ben-zhang committed:
fixed syntax

ben-zhang committed:
more syntax

ben-zhang committed:
blah

ben-zhang committed:
boost stuff
